src/main/kotlin/lotto/domain/LottoStore.kt

```diff
@@ -1,12 +1,25 @@
 package lotto.domain
 
-import lotto.domain.strategy.NumberGenerateStrategy
-
 object LottoStore {
-    const val LOTTO_PRICE = 1000
+    val LOTTO_PRICE = LottoPrice(1000)
+
+    fun purchase(
+        amount: PurchaseAmount,
+        manualCount: LottoCount,
+        shuffleStrategy: (List<LottoNumber>) -> List<LottoNumber>,
+        purchase: () -> List<LottoNumber>
+    ): LottoTickets {
+        val purchaseCount = amount / LOTTO_PRICE
+        val manual = purchaseManual(manualCount, purchase)
+        val auto = purchaseAuto(purchaseCount - manualCount, shuffleStrategy)
+        return manual.concat(auto)
+    }
+
+    fun purchaseAuto(count: LottoCount, shuffleStrategy: (List<LottoNumber>) -> List<LottoNumber>): LottoTickets {
+        return LottoTickets.create(count, shuffleStrategy)
+    }
 
-    fun purchase(amount: String, numberStrategy: NumberGenerateStrategy): LottoTickets {
-        val count = PurchaseAmount(amount).calculateLottoCount(LOTTO_PRICE)
-        return LottoTickets.create(count, numberStrategy)
+    fun purchaseManual(manualCount: LottoCount, purchase: () -> List<LottoNumber>): LottoTickets {
+        return LottoTickets(manualCount.repeat { LottoTicket(purchase.invoke().toSet()) })
```

Comment:
`invoke()`는 제거 가능해 보여요!

```suggestion
        return LottoTickets(manualCount.repeat { LottoTicket(purchase().toSet()) })
```

Reply:
`invoke()`를 제거하면 `() -> List<LottoNumber>` 타입이 되어버려서 toSet이 불가능하네요 😭